Realm/RLMSyncSubscription.h

```diff
@@ -0,0 +1,336 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2021 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import <Foundation/Foundation.h>
+
+@class RLMObjectId;
+
+#pragma mark - Subscription States
+
+typedef NS_ENUM(NSUInteger, RLMSyncSubscriptionState) {
+    /// The subscription is complete and the server has sent all the data that matched the subscription
+    /// queries at the time the subscription set was updated. The server is now in a steady-state
+    /// synchronization mode where it will stream update as they come.
+    RLMSyncSubscriptionStateComplete,
+    /// The subscription encountered an error and synchronization is paused for this Realm. You can
+    /// find the error calling error in the subscription set to get a description of the error. You can
+    /// still use the current subscription set to write a subscription.
+    RLMSyncSubscriptionStateError,
+    /// The subscription is persisted locally but not yet processed by the server, which means
+    /// the server hasn't yet returned all the data that matched the updated subscription queries.
+    RLMSyncSubscriptionStatePending,
+    /// The subscription set has been super-ceded by an updated one, this typically means that
+    /// someone is trying to write a subscription on a different instance of the subscription set.
+    /// You should not use a super-ceded subscription set and instead obtain a new instance of
+    /// the subscription set to write a subscription.
+    RLMSyncSubscriptionStateSuperceded
+};
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface RLMSyncSubscription : NSObject
+
+/// Identifier of the subscription.
+@property (nonatomic, readonly) RLMObjectId *identifier;
```

Comment:
There doesn't appear to be anything you can actually do with this identifier. I'd expect there to be a lookup function using it and maybe for adding a subscription to return the identifier.

Reply:
cc @dianaafanador3